pico/tools/src/main/java/io/helidon/pico/tools/DefaultInterceptorCreator.java

```diff
@@ -524,36 +550,139 @@ private static class ProcessorBased extends AbstractInterceptorProcessor {
+                    .forEach(ee -> result.add(
+                            create(ee, serviceLevelAnnos, interceptorAnnotationTriggers)));
+            return result;
+        }
+
+        void gatherInjectableConstructor(List<InterceptedElement> result,
+                                         Set<AnnotationAndValue> serviceLevelAnnos,
+                                         Set<String> interceptorAnnotationTriggers) {
+            serviceTypeElement.getEnclosedElements().stream()
+                    .filter(e -> e.getKind() == ElementKind.CONSTRUCTOR)
+                    .map(ExecutableElement.class::cast)
+                    .filter(ee -> !ee.getModifiers().contains(Modifier.PRIVATE))
+                    .filter(ee -> {
+                        boolean hasInject = ee.getAnnotationMirrors().stream()
+                                .map(TypeTools::createAnnotationAndValue)
+                                .map(AnnotationAndValue::typeName)
+                                .map(TypeName::name)
+                                .anyMatch(anno -> TypeNames.JAKARTA_INJECT.equals(anno) || TypeNames.JAVAX_INJECT.equals(anno));
+                        return hasInject;
+                    })
+                    .forEach(ee -> result.add(
+                            create(ee, serviceLevelAnnos, interceptorAnnotationTriggers)));
+
+            if (result.size() > 1) {
+                throw new ToolsException("There can be at most one injectable constructor for: " + serviceTypeName());
+            }
+
+            if (result.size() == 1) {
+                return;
+            }
+
+            // find the no-arg constructor as the fallback
+            serviceTypeElement.getEnclosedElements().stream()
+                    .filter(e -> e.getKind() == ElementKind.CONSTRUCTOR)
+                    .map(ExecutableElement.class::cast)
+                    .filter(ee -> !ee.getModifiers().contains(Modifier.PRIVATE))
+                    .filter(ee -> ee.getParameters().isEmpty())
+                    .forEach(ee -> result.add(
+                            create(ee, serviceLevelAnnos, interceptorAnnotationTriggers)));
+            if (result.isEmpty()) {
+                throw new ToolsException("There should either be a no-arg or injectable constructor for: " + serviceTypeName());
+            }
+        }
+
+        /**
+         * @return returns true if the given method is implemented in one of the provided interface type names
+         */
+        boolean mapsToAnInterface(ExecutableElement ee,
+                                  Set<TypeName> interfaces) {
+            for (TypeName typeName : interfaces) {
+                TypeElement te = processEnv.getElementUtils().getTypeElement(typeName.name());
+                Objects.requireNonNull(te, typeName.toString());
+                boolean hasIt = te.getEnclosedElements().stream()
+                        // _note to self_: there needs to be a better way than this!
```

Comment:
Anyone know of a better way to test if a method belongs to an interface element in annotation processing?  I couldn't find anything.